### PR TITLE
Extract `Response` into a protocol and rename code in statusCode #56

### PR DESCRIPTION
**Pull request by MP0w**

### CODE REVIEW

**Comment by joanromano** [comment]
I get the general idea, but can you expand here with another concrete example of `ResponseFieldsProvider`?

File: Source/Router.swift [hunk 1]
```diff
@@ -34,29 +34,47 @@ public struct Request {
 }
 
 /**
- A Response struct which can be used in `RouteHandlers` to provide valid responses.
+ A protocol to adopt when a Serializable Object needs to also provide response status code and/or headerFields
```
Comment (joanromano) [inline Source/Router.swift]: `Serializable object`

File: Source/Router.swift [hunk 2]
```diff
@@ -34,29 +34,47 @@ public struct Request {
...
+ You may usually use `Response` to wrap your `Serializable` object to just achieve the result or directly implement the protocol. For examply `JSONAPISerializer` implement the protocol in order to be able to provide custom status code in the response.
```
Comment (joanromano) [inline Source/Router.swift]: `For example`

**Comment by joanromano** [comment]
👍

**Comment by MP0w** [comment]
Example use case https://github.com/devlucky/Kakapo/issues/57

**Comment by MP0w** [comment]
As for now the only way to return custom code or header fields was to wrap a `Serializable` object in `Response`. Complex `CustomSerializable` object (like json API) might need this functionality without requiring to wrap the object in `Response`, but just conforming to the protocol

**Comment by MP0w** [comment]
Yep https://github.com/devlucky/Kakapo/issues/57

Inviato da iPad

**Pull request merged**